<a href="https://colab.research.google.com/github/LeeLee-ai/Lee/blob/main/list_practice.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 문제 1.
- 문자 10개를 저장할 수 있는 리스트를 만들고 10개의 문자를 입력받아 입력받은 문자를 이어서 출력하는 프로그램을 작성하시오.
- 입력 예: A B C D E F G H I J
- 출력 예: ABCDEFGHIJ

In [ ]:
lists = input('문자10개를 띄어쓰기를 해가며 입력하십시오:')
while len(lists)>19:
  print('다시 입력해주십시오')
  lists = input('문자10개를 띄어쓰기를 해가며 입력하십시오:')
lists2 = lists.split(' ')
alphabet = "".join(lists2)
print(alphabet)

문자10개를 띄어쓰기를 해가며 입력하십시오:A A A A A A A A A A A A A A A 
다시 입력해주십시오
문자10개를 띄어쓰기를 해가며 입력하십시오:A B C D E F G H I J
ABCDEFGHIJ


# 문제 2.
- 10개의 문자를 입력받아서 첫 번째 네 번째 일곱 번째 입력받은 문자를 차례로 출력하는 프로그램을 작성하시오.
- 입력 예: A B C D E F G H I J
- 출력 예: A D G

In [ ]:
lists = input('문자열을 최소 7개 이상 띄어쓰기를 하며 입력하십시오:')
while len(lists)<13:
  print('다시 입력하십시오')
  lists = input('문자열을 최소 7개 이상 띄어쓰기를 하며 입력하십시오:')
lists2 = lists.split(' ')
print(lists2[0],lists2[3],lists2[6])


문자열을 최소 7개 이상 띄어쓰기를 하며 입력하십시오:A B C D E F G H I J
A D G


# 문제 3.
- 빈 리스트을 선언하고 정수를 차례로 입력받다가 0 이 입력되면 0 을 제외하고 그 때까지 입력된 정수를 가장 나중에 입력된 정수부터 차례대로 출력하는 프로그램을 작성하시오.
- 입력 예: 
    * 3 
    * 5
    * 10 
    * 55 
    * 0
- 출력 예: 55 10 5 3

In [ ]:
lists = []
num = int(input('숫자를 입력하십시오:'))
while num>0:
  lists.append(num)
  num = int(input('숫자를 입력하십시오:'))
lists2 = lists[::-1]
print(*lists2)

숫자를 입력하십시오:3
숫자를 입력하십시오:5
숫자를 입력하십시오:10
숫자를 입력하십시오:55
숫자를 입력하십시오:0
55 10 5 3


# 문제 4.
- 1반부터 6반까지의 평균점수를 저장한 후 두 반의 반 번호를 입력받아 두 반 평균점수의 합을 출력하는 프로그램을 작성하시오.
- 반별 평균점수는 초기값으로 1반부터 차례로 85.6 79.5 83.1 80.0 78.2 75.0으로 초기화하고, 
- 출력은 소수 두 번째 자리에서 반올림하여 소수 첫째자리까지 한다.
- 입력 예: 1 3 (1반 3반을 의미)
- 출력 예: 168.7

In [ ]:
score = float(input('반 평균점수를 입력하십시오:'))
lists = []
i = 0
y = 0
while score>0 and i<5:
  lists.append(score)
  score = float(input('반 평균점수를 입력하십시오:'))
  i+=1
num = input('몇반인지 입력하십시오:')
lists2 = num.split(" ")
for e in range(0,len(lists2)):
  t = int(lists2[e])
  y+=lists[t-1]
average = y/len(lists2)
print(round(average,1))



  


반 평균점수를 입력하십시오:56.6
반 평균점수를 입력하십시오:78.8
반 평균점수를 입력하십시오:88.8
반 평균점수를 입력하십시오:99.9
반 평균점수를 입력하십시오:77.7
반 평균점수를 입력하십시오:66.6
몇반인지 입력하십시오:1 3
72.7


# 문제 5.
- 10개의 정수를 입력받아 그 중 가장 작은 수를 출력하는 프로그램을 작성하시오.
- 입력 예: 5 10 8 55 6 31 12 24 61 2
- 출력 예: 2

In [ ]:
num = input('숫자열개를 입력하십시오:')
if num.count(" ")>9:
  num = input("숫자열개를 입력하십시오")
lists = num.split(" ")
minivalue = int(lists[0])
for i in range(0,len(lists)):
  if minivalue>int(lists[i]):
    minivalue = int(lists[i])
print(minivalue)

숫자열개를 입력하십시오:5 10 8 55 6 31 12 24 61 2
2


# 문제 6.
- 10개의 정수를 입력받아 100 미만의 수 중 가장 큰 수와 100 이상의 수 중 가장 작은 수를 출력하는 프로그램을 작성하시오.
- 입력 예: 88 123 659 15 443 1 99 313 105 48
- 출력 예: 99 105

In [ ]:
num = input('정수 10개를 입력하십시오:')
if num.count(" ")>9:
  num = input('정수 10개를 입력하십시오:')
lists = list(map(int,num.split(" ")))
lists2 = []
lists3 = []
for i in lists:
  if i>=100:
    lists2.append(i)
  else:
    lists3.append(i)
minivalue = lists2[0]
for e in range(0,len(lists2)):
  if minivalue>lists2[e]:
    minivalue = lists2[e]
maxvalue = lists3[0]
for u in range(0,len(lists3)):
  if maxvalue<lists3[u]:
    maxvalue = lists3[u]
print(maxvalue)
print(minivalue)


정수 10개를 입력하십시오:88 123 659 15 443 1 99 313 105 48
99
105


# 문제 7.
- 10개의 정수를 입력받아 리스트에 저장한 후 짝수 번째 입력된 값의 합과 홀수 번째 입력된 값의 평균을 출력하는 프로그램을 작성하시오.
- 평균은 반올림하여 소수첫째자리까지 출력한다.
- 입력 예: 95 100 88 65 76 89 58 93 77 99
- 출력 예: 
    * sum : 446
    * avg : 78.8

In [ ]:
num = input('정수를 입력하십시오:')
if num.count(" ")>9:
  num = input('정수를 입력하십시오:')
lists = list(map(int,num.split(" ")))
x = 0
y = 0
e = 0
for i in range(0,len(lists)):
  if i%2!=0:
    x+=lists[i]
  elif i%2==0 or i==0:
    y+=lists[i]
    e+=1
print(x)
average = y/e
print(round(average,1))


정수를 입력하십시오:95 100 88 65 76 89 58 93 77 99
446
78.8


# 문제 8.
- 10개의 정수를 입력받아 리스트에 저장한 후 내림차순으로 정렬하여 출력하시오.
- 입력 예: 95 100 88 65 76 89 58 93 77 99
- 출력 예: 100 99 95 93 89 88 77 76 65 58

In [24]:
num = input('정수를 입력하십시오:')
lists = list(map(int,num.split(" ")))
lists2 = []
e = 0
while e<=9:
  maxvalue = lists[0]
  r = len(lists)
  for i in range(0,r):
    if maxvalue<lists[i]:
      maxvalue = lists[i]
  lists2.append(maxvalue)
  lists.remove(maxvalue)
  e+=1
print(lists2)


정수를 입력하십시오:95 100 88 65 76 89 58 93 77 99
[100, 99, 95, 93, 89, 88, 77, 76, 65, 58]
